### ISSUE TITLE
Suite exits while running cmake for libsoxr-git

### ISSUE THREAD

**Issue by Vangelis66** [jb-alvarado/media-autobuild_suite#507]
Hello 

After more than a year passed since I last successfully ran the suite to build a custom, non-free, ffmpeg-win32 static build, earlier today I created a new clean install of the suite, based on latest version, and tried to compile ffmpeg; this is on a Windows x86 laptop... 
Sadly, the script exited while being in the "Starting 32bit compilation of audio tools" section, in fact while it was running cmake for libsoxr-git: 
```
┌ libsoxr git ....................................... [Recently updated]
├ Running cmake...
Likely error:
CFLAGS: -mthreads -mtune=generic -O2 -pipe
LDFLAGS: -pipe -static-libgcc -static-libstdc++
cmake .. -G Ninja -DBUILD_SHARED_LIBS=off -DCMAKE_INSTALL_PREFIX=/local32 -DUNIX=on -DCMAKE_BUILD_TYPE=Release -DWITH_LSR_BINDINGS=off -DBUILD_TESTS=off
cmake failed. Check G:/Compiler-v3/media-autobuild_suite/build/libsoxr-git/build
This is required for other packages, so this script will exit.
```
I'd like to build ffmpeg with libsoxr support and sadly I am not an experienced suite user to try and fix such errors on my own, nor am I familiar with a Linux-y environment or commands :cry: 

> Make sure the suite is up-to-date before reporting an issue. 

It is; clean install created today from latest snapshot; and when I run media-autobuild_suite.bat, 
I see that it also checks for updated components... 

> Try running the build again at a later time.
 
This has the same result, as has deleting folder /build/libsoxr-git and trying anew...

> Creating diagnostics file...

As per console instructions, here it is: 
[logs.zip](https://0x0.st/x6j.zip)

This is from the first time the error occured, not from later attempts...
And I suppose this has nothing to do with reported error, but in fact am trying to build latest 3.3 tag release, so have modified line 1155 of media-suite_compile.sh to read
`if do_vcs "https://git.ffmpeg.org/ffmpeg.git#tag=n3.3"; then`

I will appreciate any hints/advice on how to overcome this hurdle...
As always, many thanks to the suite maintainer and contributors...
Regards.

**Comment by wiiaboo** [comment]
Did you try deleting build\libsoxr-git? There's probably a lot that's changed in a year of not running the suite.

**Comment by Vangelis66** [comment]
@wiiaboo wrote: 

> Did you try deleting build\libsoxr-git?

Yes, please read my report carefully: 

>> This has the same result, as has deleting folder /build/libsoxr-git and trying anew... 

> There's probably a lot that's changed in a year of not running the suite.

**Probably** is an understatement, I guess, but as reported this is on a clean new install, where everything is being built up from scratch; I just mentioned I last used the suite more than a year ago (March 2016) to illustrate I am unfamiliar with the suite/libs/ffmpeg changes since then; FWIW, on this same machine, libsoxr-git built fine back then...

**Comment by wiiaboo** [comment]
Seems like something is wrong with CMake on your installation, so it's not a bug with the suite.
Open mintty and run `cmake --version` and then
```
cd /build/libsoxr-git
mkdir build-32
cd build-32
cmake .. -G Ninja -DBUILD_SHARED_LIBS=off -DCMAKE_INSTALL_PREFIX=/local32 -DUNIX=on -DCMAKE_BUILD_TYPE=Release -DWITH_LSR_BINDINGS=off -DBUILD_TESTS=off
```
If there's some error being printed thath could help. If not, probably report CMake being broken in https://github.com/Alexpux/MSYS2-packages.

**Comment by wiiaboo** [comment]
Also, 32-bit is starting to be deprecated and forgotten in a lot of places. I'd suggest moving on to a more recent OS with 64-bit, if your CPU supports it.

**Comment by Vangelis66** [comment]
> Seems like something is wrong with CMake on your installation, 
> so it's not a bug with the suite.
> Open mintty and run cmake --version and then

I am afraid this is very bad news for me...
```
$ cmake --version
bash: cmake: command not found
```
Additionally, trying manually to run /msys32/mingw32/bin/cmake.exe produces: 

![cmake](https://cloud.githubusercontent.com/assets/9669492/25280601/3979e752-26b3-11e7-89b9-a60a5ed5e861.jpg)

**Comment by wiiaboo** [comment]
https://msdn.microsoft.com/en-us/library/windows/desktop/dd433795(v=vs.85).aspx
Yeah, that means you'd need at least Windows 7.

**Comment by wiiaboo** [comment]
Windows Vista extended support ended 11th of April this year and mainstream support ended all the way back in 2012, btw.

**Comment by Vangelis66** [comment]
@wiiaboo, I truly appreciate your attention to my issue, which indeed turned out to be 
a combination of latest mingw32 toolchain + Vista SP2 x86 OS, so not a suite-scripts issue per se.
I have always valued your help offered in the past, plus I see you are still doing a sterling job wrt the suite :+1: 

But with respect, I am fully aware of my OS being EOL'ed by M$; for crying out loud, it's been only 10 days 
since then, yet many app authors were anxious to drop Vista like a hot potato many months prior. 
XP has been EOL'ed since like April 2014, yet enough software still has support for it; why did Vista had to die such a horrible death?

Returning to FFmpeg itself, pure FFmpeg code continues to support XP & Vista, but some of the libs have already moved onto 7+ only code; as a result, the free Zeranoe automated builds won't run on XP since the end of Feb 2016 and on Vista since mid-July 2016 (see my post and discussion [here](https://github.com/streamlink/streamlink/pull/224#issuecomment-263467807), plus plenty of talk in the Zeranoe Forums...). 

Thanks to various good people on the net, I can still (but with difficulty) find an XP/Vista compatible FFmpeg win32 binary, but in the back of my mind I had always counted on media-autobuild_suite to be a last resort for compiling myself; imagine my huge let down today when I discovered this is no longer possible :sob:

But despite me not being very techie when it comes to compilers, I don't like to easily give up in general; especially since after some preliminary tests it looks as though it is only cmake of the whole toolchain that is Vista incompatible. 
My first approach was to revisit my March 2016 image of the suite, where I get: 
```
$ cmake --version
cmake version 3.4.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
Then I thought of copying cmake.exe 3.4.1 (from old image) and putting it inside 
`.\media-autobuild_suite\msys32\mingw32\bin`, backing-up first newer version 3.7.1 
that comes with the updated toolchain, but alas I got the same error; I have no clue, 
but possibly mingw32's cmake.exe is a shared binary depending on other Vista incompatible DLLs ? 

I changed approach by visiting https://www.kitware.com/platforms/#cmake and then 
https://cmake.org/download/ ; latest version is 3.8, of which they provide [pre-compiled win32 binaries](https://cmake.org/files/v3.8/cmake-3.8.0-win32-x86.zip) ; these are standalone binaries compiled with Visual Studio 2013, so not dependent on mingw32; unpacked, made a back up of existing same-name files within `.\media-autobuild_suite\msys32\mingw32\bin` and then threw in the 3.8 ones, initially inside `.\cmake-3.8.0-win32-x86\bin`; 
I also had to merge `.\cmake-3.8.0-win32-x86\share` with `.\media-autobuild_suite\msys32\mingw32\share` to omit an error about cmake not being correctly installed...
AFAICT, all is fine now with cmake:
```
$ cmake --version
cmake version 3.8.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
Firing up the suite, I got past the initial cmake error that prompted me to file the original issue: 
```
  Running git update for libsoxr...
β” libsoxr git ....................................... [Recently updated]
β” Running cmake...
β” Running install...
β” Stripping...
β”” libsoxr-git ................................................ [Updated]
```
Currently the script is still running, having successfully compiled 
`fdkaac.exe, flac.exe, lame.exe, metaflac.exe, oggdec.exe, oggenc.exe, opusenc.exe, opusdec.exe, opusinfo.exe, speexdec.exe, speexenc.exe, cwebp.exe, dwebp.exe, fribidi.exe, img2webp.exe, ocspcheck.exe, openssl.exe, webpinfo.exe, webpmux.exe, kvazaar.exe, MP4Box.exe, rtmpdump.exe (and friends...), vpxdec.exe, vpxenc.exe` 
and is currently trying to build x264.exe with ffms2+liblsmash support (taking a lot of time doing so...) 
Will report back how things progressed... 

To recap: 
1. It appears, with a good level of certainty, that the reason compilation fails on Vista SP2 x86 (fully updated) is Vista incompatible cmake tool provided by the msys2/mingw32 toolchain, all other tools seem to be working as expected.
2. Manually replacing provided cmake 3.7.1 with a Vista compatible 3.8 version, as outlined previously, appears to have fixed breakage.

Do you think I should report this to [MSYS2-packages/issues](https://github.com/Alexpux/MSYS2-packages/issues) or are there absolutely 0 chances of Vista compatibility being resumed? I find it weird that the maintainers of CMAKE still support Vista, but the MSYS2-package one doesn't (???). 
And, by all likelihood, when the toolchain gets updated by pacman I shall have to again "patch" cmake, 
is that right? 

Apologies if this comment has turned into a voluminous guide, but it was meant for fellow Vista users; 
they may account for ca. 1.01% of desktop OSes, but they are die-hards...

UPDATE: Currently at 
```
β” ffmpeg git ........................................ [Recently updated]
β” Changing options to comply to nonfree...
β” Compiling static FFmpeg...
β” Running configure...
```
so fingers crossed in about 70 - 80 mins (old CPU) I'll have my FFmpeg-3.3-win32 non-free binary.

EDIT: From what I've seen so far, it is only **libsoxr** & **x265.exe/libx265** that make use of **cmake**; 
so `#--enable-libsoxr #--enable-libx265` (in ffmpeg_options.txt) and `x2652=2` (in media-autobuild_suite.ini) will make compiling on Vista possible, hopefully...

**Comment by wiiaboo** [comment]
> Do you think I should report this to MSYS2-packages/issues or are there absolutely 0 chances of Vista compatibility being resumed? I find it weird that the maintainers of CMAKE still support Vista, but the MSYS2-package one doesn't (???).

Can't harm asking. You can also install a previous version of cmake and block updates to it, but that's fragile. Better to ask if possible to maintain support, unless it's preventing new features from being included. In that case, I very much doubt they care to support Vista.

Windows 7 is altogether a better version of Vista with nothing worse afaik, what's keeping you from updating?

**Comment by Vangelis66** [comment]
I said 

> so fingers crossed in about 70 - 80 mins (old CPU) I'll have my FFmpeg-3.3-win32 non-free binary.

... :cry: I seem to have challenged the Gods and spoken too soon...  
Right there at the end, after hours of successful compilation, I got a fatal error: 
```
β” ffmpeg git ........................................ [Recently updated]
β” Changing options to comply to nonfree...
β” Compiling static FFmpeg...
β” Running configure...
β” Running make...
Likely error:
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x70f): undefined reference to `png_write_image'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x71e): undefined reference to `png_write_end'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x75d): undefined reference to `png_set_tRNS'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x1afd): undefined reference to `png_create_write_struct'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x1b0d): undefined reference to `png_create_info_struct'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x1b6e): undefined reference to `png_destroy_write_struct'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x1be4): undefined reference to `png_destroy_write_struct'
G:/Compiler-v3/media-autobuild_suite/local32/lib/libzvbi.a(libzvbiinline_la-exp-gfx.o):(.text+0x1c25): undefined reference to `png_destroy_write_struct'
collect2.exe: error: ld returned 1 exit status
make: *** [/build/ffmpeg-git/Makefile:136: ffmpeg_g.exe] Error 1
make failed. Check G:/Compiler-v3/media-autobuild_suite/build/ffmpeg-git/build-static-32bit/ab-suite.make.log
This is required for other packages, so this script will exit.
```
AFAICT, it doesn't appear Vista related, but rather an issue caused by **libzvbi**; I am open to further suggestions; I was so sure things will succeed this time around that I left the machine unattended 
hoping to find compiled ffmpeg when I returned; instead this fatal error... 
Why is it so hard to get what one wants?
diagnostics file: [logs.zip](https://0x0.st/xUp.zip)

> Windows 7 is altogether a better version of Vista with nothing worse afaik, 
> what's keeping you from updating?

It's hard for others to believe, but I actually prefer Vista to 7, certainly over 8+. 
But the main reason is this is an old laptop which has been very well kept but  
has also been serviced a number of times (screen, HDD changed last year, RAM increased to 3GB), 
but at this time I am unwilling to invest more on a OS change... MS most certainly 
doesn't provide legit Win7 keys any more, so one will have to hunt ones on ebay or the like...

From a financial standpoint, things are worse than dire here in Greece, my goal is to wait 
for at least a year until I save up (doubtful) to buy a new laptop with modern hardware 
(with Win10 being the only option in the market currently).  
I am not a fan of Win10 either, considering the learning curve one 
coming from Vista/7 has to go through to master it...
So Vista it is for some months more...

Cheers

PS: Do you feel I should submit a separate issue wrt this recent failure?

**Comment by wiiaboo** [comment]
That error is probably because I never tested compiling FFmpeg with zvbi without some other lib including libpng. No need to open an issue about it.

**Comment by Vangelis66** [comment]
... Thanks, but what is the course of action to follow now? 
Just `#--enable-libzvbi` in the options? 
I don't see an `--enable-libpng` option either...

FWIW, in my last suite-compiled FFmpeg.exe (Mar 2016) , `--enable-libzvbi` worked OK (without libpng): 
``` 
ffmpeg version N-78991-g1a6a08f-gcaeed04 Copyright (c) 2000-2016 the FFmpeg deve
lopers
built with gcc 5.3.0 (Rev2, Built by MSYS2 project)
configuration:  --disable-w32threads --enable-fontconfig --enable-libass --enabl
e-libbs2b --enable-libfreetype --enable-libfribidi --enable-libkvazaar --enable-
libmp3lame --enable-libopus --enable-libsoxr --enable-libspeex --enable-libtwola
me --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable
-libx265 --enable-libxvid --enable-libzvbi --enable-nonfree --enable-openssl --e
nable-libilbc --enable-libwavpack --enable-libfaac --enable-libfdk-aac --enable-
libgsm --enable-libtheora --enable-libvo-amrwbenc --disable-encoder=vorbis --ena
ble-libxavs --enable-librtmp --enable-libssh --enable-doc --disable-ffplay --dis
able-ffprobe --enable-gpl --enable-version3 --disable-debug
libavutil      55. 19.100 / 55. 19.100
libavcodec     57. 28.100 / 57. 28.100
libavformat    57. 28.100 / 57. 28.100
libavdevice    57.  0.101 / 57.  0.101
libavfilter     6. 39.102 /  6. 39.102
libswscale      4.  0.100 /  4.  0.100
libswresample   2.  0.101 /  2.  0.101
libpostproc    54.  0.100 / 54.  0.100
```
Again, a mil "thank you"s for staying on with me thus far... 
I pray I can get this thing resolved, one way or another...

**Comment by wiiaboo** [comment]
It's fixed. Just update the suite.

**Comment by Vangelis66** [comment]
... All my prayers and well-wishing go towards you right now: 
```
ffmpeg version n3.3-g4925537004+39 Copyright (c) 2000-2017 the FFmpeg developers

  built with gcc 6.3.0 (Rev2, Built by MSYS2 project)
  configuration:  --enable-libmp3lame --enable-libopus --enable-libvorbis --enab
le-libvpx --enable-libx264 --enable-libx265 --enable-libass --enable-libbs2b --e
nable-libgsm --enable-libilbc --enable-librtmp --enable-libsoxr --enable-libspee
x --enable-libtheora --enable-libtwolame --enable-libwavpack --enable-libwebp --
enable-libxavs --enable-libxvid --enable-openssl --enable-libfdk-aac --enable-li
bkvazaar --enable-libssh --enable-libzvbi --enable-libopenh264 --disable-encoder
=vorbis --enable-doc --disable-ffplay --disable-ffprobe --extra-cflags=-DLIBTWOL
AME_STATIC --extra-cflags=-fopenmp --extra-libs=-lgomp --extra-cflags=-DLIBSSH_S
TATIC --extra-ldflags='-Wl,--allow-multiple-definition' --extra-libs=-lpng --ext
ra-cflags=-DKVZ_STATIC_LIB --enable-gpl --enable-nonfree --disable-debug --disab
le-sdl2
  libavutil      55. 58.100 / 55. 58.100
  libavcodec     57. 89.100 / 57. 89.100
  libavformat    57. 71.100 / 57. 71.100
  libavdevice    57.  6.100 / 57.  6.100
  libavfilter     6. 82.100 /  6. 82.100
  libswscale      4.  6.100 /  4.  6.100
  libswresample   2.  7.100 /  2.  7.100
  libpostproc    54.  5.100 / 54.  5.100
Hyper fast Audio and Video encoder
```
It wouldn't have been possible without your invaluable guidance/help :+1: 

I originally planned to build an **FFmpeg-3.3-release** binary, based on 3.3 source 
provided either [here](http://ffmpeg.org/releases/ffmpeg-3.3.tar.bz2) or [here](https://github.com/FFmpeg/FFmpeg/releases/tag/n3.3). 
The GitHub link as well as [tag n3.3](https://git.ffmpeg.org/gitweb/ffmpeg.git/tag/8a5f79827600f88f9490c62a5ba7a3f71bc55e50) both suggest n3.3 is based on https://github.com/FFmpeg/FFmpeg/commit/1968a1eef1cae22e162259d7082c2eea98d81e32
I had consulted a Mar 2016 [post](https://github.com/jb-alvarado/media-autobuild_suite/issues/304#issuecomment-198599857) of yours which advised: 
```
Compiling FFmpeg releases is just a matter of adding #tag=n3.0 at the end of FFmpeg's git url.
# 3.0 release
# source as was used for release 3.0
do_vcs "https://git.videolan.org/git/ffmpeg.git#tag=n3.0"
```
so I went ahead and modified [line 1156](https://github.com/jb-alvarado/media-autobuild_suite/blob/e7233aa62289c107093f3c63280207163647c487/build/media-suite_compile.sh#L1156) to read 
`if do_vcs "https://git.ffmpeg.org/ffmpeg.git#tag=n3.3"; then`
with the expectation it would compile tag n3.3.
_However_, my end compile (many thanks again) declares: 
`ffmpeg version n3.3-g4925537004+39`, not `ffmpeg version n3.3`
Upon searching, [commit git-4925537](https://github.com/FFmpeg/FFmpeg/commit/4925537) appears to have been pushed just 7 hours ago, so something's gone amiss in my settings... 
How have I messed things up and what exactly am I to issue and where (in layman's terms) to compile the 3.3 source? 

Apologies for being such a persistent nuisance, at least I show up maybe once a year, LOL!

Regards

**Comment by wiiaboo** [comment]
That's just the version from https://github.com/jb-alvarado/media-autobuild_suite/blob/e7233aa62289c107093f3c63280207163647c487/build/media-suite_compile.sh#L1161-L1163.
You should've commented these lines or changed "origin/master" to "origin/n3.3".

The actual compiled code is the same as 3.3.
